### ISSUE TITLE
Font option is slightly confusing

### ISSUE THREAD

**Issue by kparal** [CorsixTH/CorsixTH#1914]
**What feature or enhancement would you like to add?**
In _Settings -> Folders -> Font_, it seems the game lets you select a different font. But it doesn't. If you select some font, "nothing happens". **Until** you select a different language, and not just any language, but a language that is not supported by the default font. This confused me when I first used this option. Concretely I thought:
1) Why has nothing happened? Do I need to restart the game? (Hint: Restarting the game doesn't change anything).
2) How can I go back "to the defaults" and take back the selected font? Even if "nothing happened", I don't want to take anything risks, how do I remove the selected font path? (Hint: You can't).

It would be more obvious if the option wasn't called `Font`, but `Font for alternative languages` or `Alternative font` or something similar, and the tooltip would be updated to describe this more clearly (i.e. the default font is used, and can't be changed, for languages supported by it, and the alternative font is used for any other languages).

**What version of CorsixTH are you using?**
0.65-beta2, Flathub

**Comment by TheCycoONE** [comment]
If you were interested in submitting a PR, this is a change to: https://github.com/CorsixTH/CorsixTH/blob/6aea7c0a943cc62da5fd74b998a527286dac5cc9/CorsixTH/Lua/languages/english.lua#L488 and https://github.com/CorsixTH/CorsixTH/blob/6aea7c0a943cc62da5fd74b998a527286dac5cc9/CorsixTH/Lua/languages/english.lua#L503

As long as it fits in the current UI, it should be minor enough to sneak into 0.65

The behaviour of this option will likely change when #1901 is addressed, but it's a reasonable stopgap measure.

**Comment by kparal** [comment]
Unfortunately my life balance doesn't allow me to become a code contributor at this moment :disappointed:

**Comment by Patil2099** [comment]
I would love to help in solving this issue @TheCycoONE. Would you guide me a little in solving this?

**Comment by lewri** [comment]
> I would love to help in solving this issue @TheCycoONE. Would you guide me a little in solving this?
I believe we just need to make those two English strings clearer for people. By current UI it just means making sure any change won't cause badly placed text.

You can test a change on those strings by editing the copy of English.lua in your install. (Found in Lua > Languages > english.lua)

**Comment by mefistotelis** [comment]
I don't think it's just a matter of two strings.

My experience:
* Installed CorixTH
* Noticed there is no "polish" in the language list
* Found old fan translation of the original game, installed it on data folder
* The game got semi-translated, but it didn't trusted the DAT/TAB fonts for some reason, so all national characters were just "?"
* My conclusion was that the project has no polish translation, and that the project also cannot properly handle original data files

How my experience could've been better:
* Show the polish language on the list, **always** allow the player to select it
* When selecting the language, show an error message "_Cannot switch language: The language you've selected contains national characters which cannot be displayed by fonts from the data folder. You need to provide a path to Unicode fonts in order to switch to that language. You can get the unicode fonts (point to a location)._"

This way I would know what to do without googling or searching, and I wouldn't push with the old translation concluding CorsixTH doesn't have one.

(also, would be nice if the game was able to handle any DAT/TAB files, not only the exact ones from original game, but that's a separate subject)

**Comment by Alberth289346** [comment]
Feel free to make a PR

**Comment by lewri** [comment]
For the other issue of importing the raw .dat it's a little more complicated than that.
It could be using a different encoding that we haven't implemented yet. Not impossible, not also favourable in the long run to allow anyone to freely try messing around with dat files.

I'm not objecting to showing all languages, but some languages need a unicode font to even make it readable. If you haven't selected it first, how would you expect it to be understood that it is language x?

**Comment by lewri** [comment]
Additional note:
Polish actually renders fine in most fonts (case-in-point: Comic Sans). But at the bottom of this image you see the issue.
![image](https://user-images.githubusercontent.com/20030128/148298062-4a7b15f6-be81-437c-ae54-20864fcb151e.png)

**Comment by mefistotelis** [comment]
> For the other issue of importing the raw .dat [...] It could be using a different encoding that we haven't implemented yet.

Yes, that's true - it actually **must** be using some kind of special encoding for the original game to display national chars.
And if the game converts the strings to unicode, and then converts them back to be displayed - this special code page would have to be implemented.

There never was a reason to convert the sting to unicode in the first place, but if this is where we are - would probably be complicated to eliminate the conversion.

> I'm not objecting to showing all languages, but some languages need a unicode font to even make it readable. If you haven't selected it first, how would you expect it to be understood that it is language x?

Then we should display that using closest English letters.
All the codepage conversion libraries today support such function. I's so standard, I actually used `unidecode.unidecode(name)` yesterday in a Python script.

(While at converting chars - if we're going to convert case of language-specific chars, we should also do it right. And if we don't want to depend on any conversion library, then all string should be just displayed without any conversion. The current look of diseases encyclopedia is very poor, with national chars being lower case and everything else converted to upper.)

**Comment by TheCycoONE** [comment]
Not all of the translations use the same encodings or sprite sheets for fonts. We only support the common code page 437 variant of the game at the moment. Anything that may use characters outside of that is unicode. If a translation doesn't need characters outside of that it may be erroneously classified as unicode.

**Comment by TheCycoONE** [comment]
I advise whoever takes this as a good first issue not get carried away with scope creep. We quickly went from a very basic implementation that could be done in a single sitting to a much larger effort and I'm afraid best is the enemy of better here.

**Comment by lewri** [comment]
@tobylane would #2398 provide some solution to this one? Maybe not completely but it would mean we cover more languages.

I think if that PR is good enough to address the majority of this then the string change in the first comment should be good enough to cover the rest.

**Comment by tobylane** [comment]
Agreed. The new text may depend on whether we say a release (or any non-dev build) must have a bundled font in it, or a system font in the linux package.

**Comment by lewri** [comment]
Just to note we have now properly implemented some improvements to unicode font detection and defaults.
In Windows and Linux AppImage (not sure about Flatpak) the CorsixTHUnicode font file is loaded if no custom font is set (a rename of the Noto Kurrent font).

We also now grey out unavailable languages and show their english translations too. 

I am not too sure what remaining improvements will be wanted here.

**Comment by ARGAMX** [comment]
We still have a bug where user need to restart the game for a newly selected font to take effect. 
But I don't think this issue is much related to this issue.

Also, as mentioned above, the following has been done:

When hovering over the Font option, show a tooltip explaining the role of the font selection.

If the font wasn't installed with the game and for some reason wasn't found and is ultimately unavailable, then display the language names in cp437 and, when hovering over a language, show a tooltip explaining that you need to select the font for the language to be available.

I think the core of this issue has been addressed and fixed.
There are still a few rough edges, but it seems better to create separate, more specific issue if needed.

This is a very general issue, and I agree it's mostly resolved already.

**Comment by TheCycoONE** [comment]
If we open a bug for that particular issue, it's within the realm of things I know how to fix.
